src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1600,12 +1495,15 @@ public static void SetTestHook(string property, bool value)
         }
     }
 
-#if CORECLR && !UNIX
+#if !UNIX
     /// <summary>
     /// Helper to start process using ShellExecuteEx. This is used only in PowerShell Core on Full Windows.
     /// </summary>
     internal class ShellExecuteHelper
     {
+        private static bool _succeeded;
+        private static int _errorCode;
```

Comment:
These 2 static fields will cause race condition when multiple Runsapces are running `ShellExecuteHelper.Start` at the same time. Using a [helper class](https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Process.cs,2839) like in .NET implementation may be a better approach.